src/execution/execution_test.c

```diff
@@ -0,0 +1,45 @@
+
+// TODO refacto run_kata to take the I/O tools in param
+
+#include "stdlib.h"
+#include "stdio.h"
+#include "test_utils.h"
+#include "execution.h"
+
+#define BUFFER_SIZE 512
+
+static test_result all_tests(void);
+
+
+int main(void) {
+    const char * result = all_tests();
+    if(result == NULL) {
+        printf("ALL TESTS PASSED\n");
+    } else {
+        printf("%s\n", result);
+    }
+
+    return EXIT_SUCCESS;
+}
+
+sized_string_t buffer = {.str = NULL, .len = 0};
+
+void before_each(void) {
+    buffer = new_sized_string_of_length(BUFFER_SIZE);
+}
+
+void tear_down(void) {
+    free_sized_string(&buffer);
+}
+
+static test_result should_get_file_not_found_if_file_does_not_exists(void);
```

Comment:
We could remove it, but. it defines better encapsulation because "other files cannot reference them during the linking phase of the compilation process. [...] they also have a fixed lifetime during the execution of the program"
As I want the tests completely independent from the rest of the project, it asserts that is the case.